### PR TITLE
#71 HttpClient can't be extended

### PR DESCRIPTION
**Pull request by vileyne**

### CODE REVIEW

File: code/lightna/lightna-engine/js/lib/HttpClient.js [hunk 1]
```diff
@@ -68,33 +84,10 @@ class HttpClient {
...
-        return this.fetch(url, {
-            ...options,
-            method: 'GET',
-        });
-    }
-
-    post(url, data, options = {}) {
-        if (!data.form_key) {
-            data.form_key = FormKey.get();
-        }
```
Comment (vesan-83) [inline code/lightna/lightna-engine/js/lib/HttpClient.js:81]: form key actually is also magento dependency, we magento need also to make it by some extend. could you make a ticket for it?

File: code/lightna/lightna-engine/js/lib/Blocks.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { $ } from 'lightna/lightna-engine/lib/utils/dom';
-import request from 'lightna/lightna-engine/lib/HttpClient';
+import { HttpClient } from 'lightna/lightna-engine/lib/HttpClient';
```
Comment (vesan-83) [inline code/lightna/lightna-engine/js/lib/Blocks.js:2]: just an idea/question, maybe make sense to call it shorter Http?

**Review by vesan-83 — APPROVED** [review]

File: code/lightna/lightna-engine/js/lib/Blocks.js (hunk 2 above)
Comment (vileyne) [reply code/lightna/lightna-engine/js/lib/Blocks.js:2]: maybe Request than?

Comment (vesan-83) [reply code/lightna/lightna-engine/js/lib/Blocks.js:2]: Sure, why not

File: code/lightna/lightna-engine/js/lib/Request.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        return responseJson;
+    }
+
+    static _onSuccess(response) {
+    }
+
+    static _onError(response) {
+    }
```
Comment (vesan-83) [inline code/lightna/lightna-engine/js/lib/Request.js:80]: in general looks good, please merge. Additionally I see one more not urgent ticket, Request should support possibility to  pass custom onSuccess and onError as argument, probably also onComplete|onFinally (called even if error happened). All known libs support it, so we might also need it

Comment (vesan-83) [reply code/lightna/lightna-engine/js/lib/Request.js:80]: custom onSuccess doesn't replace default onSuccess in class, but is called addly

Comment (vesan-83) [reply code/lightna/lightna-engine/js/lib/Request.js:80]: ok, maybe we don't need it

**Pull request merged**